src/shared/lib/AlterationsUtils.ts

```diff
@@ -0,0 +1,78 @@
+export type FilteredOutAlterations = {
+    /**
+     * how many alterations are not shown
+     */
+    notShownAlteredCases: number;
+    /**
+     * number of genes associated with the hidden alterations
+     */
+    affectedGenes: number;
+};
+
+/**
+ * Returns true if the left map contains all genes from the right map with the same number of correspondent alterations.
+ * @param leftAlteredCasesPerGene superset candidate
+ * @param rightAteredCasesPerGene subset candidate
+ */
+export function subset(
+    leftAlteredCasesPerGene: Map<string, number>,
+    rightAteredCasesPerGene: Map<string, number>
+): boolean {
+    const leftGenes = new Set(leftAlteredCasesPerGene.keys());
+    for (const rightGene of rightAteredCasesPerGene.keys()) {
+        if (!leftGenes.has(rightGene)) {
+            return false;
+        }
+        if (
+            leftAlteredCasesPerGene.get(rightGene)! <
+            rightAteredCasesPerGene.get(rightGene)!
+        ) {
+            return false;
+        }
+    }
+    return true;
+}
+
+/**
+ * Counts how many alterations missing in how many genes
+ * @param leftAlteredCasesPerGene reference
+ * @param rightAteredCasesPerGene alterations to count
+ */
+export function computedFilteredOutAlterations(
+    leftAlteredCasesPerGene: Map<string, number>,
+    rightAteredCasesPerGene: Map<string, number>
```

Comment:
`Altered`

Reply:
also I think this is a bit hard to understand, it would be better to name the parameters something thats more relevant to their typical use case, rather than just left and right

Reply:
This file was dropped.